### ISSUE TITLE
[BUG] pm2 flush <app_id> flushes all the logs

### ISSUE THREAD

**Issue by alielkhateeb** [Unitech/pm2#4074]
## What's going wrong?
When I ran `pm2 flush <app_id>` pm2 flushed all the logs for all 10 apps I have.

## Supporting information
PM2 REPORT (Fri Dec 14 2018 10:11:38 GMT+0000 (UTC))

--- Daemon -------------------------------------------------

pm2d version         : 3.1.2
node version         : 9.4.0
node path            : /usr/local/bin/pm2
argv                 : /usr/local/bin/node,/usr/local/lib/node_modules/pm2/lib/Daemon.js
argv0                : node
user                 : root
uid                  : 0
gid                  : 0
uptime               : 3min

--- CLI ----------------------------------------------------

local pm2            : 3.1.2
node version         : 9.4.0
node path            : /usr/local/bin/pm2
argv                 : /usr/local/bin/node,/usr/local/bin/pm2,report
argv0                : node
user                 : root
uid                  : 0
gid                  : 0

--- System info --------------------------------------------

arch                 : x64
platform             : linux
type                 : Linux
cpus                 : Intel(R) Xeon(R) CPU @ 2.50GHz
cpus nb              : 2
freemem              : 1115119616
totalmem             : 7864020992
home                 : /root

**Comment by Unitech** [comment]
published

```bash
$ npm install pm2@latest -g
$ pm2 update
```

**Comment by alielkhateeb** [comment]
Hi, I am on pm2 3.3.1, and now when I run `pm2 flush <app_id>` it doesn't even flush any of the apps.

**Comment by ToderitaGeorgeCristian** [comment]
I am on 4.2.1 and still doesn't delete logs.
pm2 flush <id/name> does not work

**Comment by nepeky4e69ka** [comment]
4.4.0 
`su - gitlab-runner -c "pm2 flush 5"  `
now working

**Comment by mirismaili** [comment]
4.4.1:

```bash
$ pm2 ls
┌────┬────────────────────┬──────────┬──────┬───────────┬──────────┬──────────┐
│ id │ name               │ mode     │ ↺    │ status    │ cpu      │ memory   │
├────┼────────────────────┼──────────┼──────┼───────────┼──────────┼──────────┤
│ 0  │ main               │ fork     │ 3    │ stopped   │ 0%       │ 0b       │
└────┴────────────────────┴──────────┴──────┴───────────┴──────────┴──────────┘

$ pm2 flush 0
[PM2] Logs flushed           # !!!!!!!!!!!!!!!!!!!!!!!!!!!!

$ pm2 flush main             # OK
[PM2] Flushing:
[PM2] /home/fedora/.pm2/logs/main-out.log
[PM2] /home/fedora/.pm2/logs/main-error.log
[PM2] Logs flushed
```

**Comment by andrewlorenz** [comment]
this is either not fixed, or the fix fell out of later versions!  I was on 4.4.0 and it doesn't work, updated to 4.5.0 and it still doesn't work!  The only thing that DOES work is "pm2 flush" which of course flushes out (empties) logs for ALL running pm2 apps.

**Comment by jeffrson** [comment]
Please reopen - this issue is still present in 5.2.0
